src/location/mod.rs

```diff
@@ -387,8 +387,20 @@ impl ObPartitionEntry {
         &self,
         part_id: i64,
     ) -> Option<&ObPartitionLocation> {
+        // logic_id = part_id in partition one
         self.parititon_location.get(&part_id)
     }
+
+    pub fn get_sub_partition_location_with_part_id(
+        &self,
+        part_id: i64,
+        sub_part_nums: i32,
+    ) -> Option<&ObPartitionLocation> {
+        // only for template sub partition
```

Comment:
what mean "only for template sub partition"

Reply:
Template sub-partition means that this kind of partition could not add or delete a partition.
Observer supports other sub-partition methods that users could add partitions or reduce partitions, but obkv-client-rust doesn't support this feature yet.